### ISSUE TITLE
Add support for scoped signing keys

### ISSUE THREAD

**Issue by darkwatchuk** [nats-io/jwt.net#14]
### Proposed change

Hi, 

I might be missing something obvious, but I can't see how to enable scoped signing keys for an account.

SigningKeys on NatsAccountClaims is only a string[] whilst nsc has the ability to set signing keys as an array of objects as well.

e.g.
```

 "signing_keys": [
      {
        "kind": "user_scope",
        "key": "AC*************************************************",
        "role": "chat_user",
        "template": {
          "pub": {},
          "sub": {}
        }
      }
    ],
```

### Use case

To be able to define and edit account signing keys  

e.g. https://www.youtube.com/watch?v=KmGtnFxHnVA

**Comment by darkwatchuk** [comment]
Didn't see that in there, but no I don't think so.  What's been added there looks to be the ability of allowing a User to **_use_** scoped signing keys rather than allowing an Account to **_have_** scoped signing keys.   It could be that the signing_keys field needs to support both a string[] and object[] too.

**Comment by darkwatchuk** [comment]
The Go library doesn't use a specific simple string[] for signing_keys on account claims, instead it references signingkeys.go

https://github.com/nats-io/jwt/blob/41dad79143e9cff55e4bde8af4df7a40ad49786c/v2/account_claims.go#L228

https://github.com/nats-io/jwt/blob/41dad79143e9cff55e4bde8af4df7a40ad49786c/v2/signingkeys.go

Maybe this library should use something like ISigningKey[] for NatsAccountClaims.SigningKeys instead of string[] before initial release and to avoid breaking the interface in future in order to support different types of signing key?

May be a basic SigningKey class to start, with implicit string convert operators.  Then a ScopedSigningKey afterwards?  Adding a custom JsonConverter for ISigningKey[] which reads/writes different json based on type would then be the way to go?

**Comment by darkwatchuk** [comment]
May be something similar to this?

```

var scopedSigningKey = new ScopedSigningKey() 
{
    Permissions = "example"
};

var t1 = new AccountClaimsMock { Id = 1, SigningKeys = ["SAxxxxxx", scopedSigningKey] };

var t1Json = JsonSerializer.Serialize(t1, new JsonSerializerOptions { WriteIndented = true });

Console.WriteLine(t1Json);




public class SigningKeyJsonConverter : JsonConverter<SigningKey[]>
{
    
    public override SigningKey[] Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
    {
        throw new NotImplementedException();        
    }

    public override void Write(Utf8JsonWriter writer, SigningKey[] signingKeys, JsonSerializerOptions options) {

        writer.WriteStartArray();
        foreach (var sk in signingKeys)
        {
            if (sk.GetType() == typeof(SigningKey))
            {
                writer.WriteStringValue(sk.ToString());
            }
            else if (sk.GetType() == typeof(ScopedSigningKey))
            {
                JsonSerializer.Serialize(writer, (ScopedSigningKey)sk);               
            }
        }
        writer.WriteEndArray();
       
    }
        
}

public class AccountClaimsMock
{
    [JsonPropertyName("fake_id")]
    public int Id { get; set; }

    [JsonPropertyName("signing_keys")]
    [JsonConverter(typeof(SigningKeyJsonConverter))]
    public SigningKey[] SigningKeys { get; set; }
}


public class ScopedSigningKey : SigningKey
{
    [JsonPropertyName("perms")]
    public string Permissions { get; set; }

}

public class SigningKey 
{
    private string _signingKey;

    public SigningKey()
    {

    }

    public override string ToString()
    {
        return _signingKey;
    }
   
    public static implicit operator string(SigningKey sk)
    {
        return sk.ToString();
    }

    public static implicit operator SigningKey(string value)
    {
        return new SigningKey() { _signingKey = value };
    }
}
```

**Comment by mtmk** [comment]
resolved by #18
